api/app/src/Command/ProcessLayCSVCommand.php

```diff
@@ -0,0 +1,206 @@
+
+    private const CHUNK_SIZE = 50;
+    
+    protected const EXPECTED_COLUMNS = [
+        'Case',
+        'ClientSurname',
+        'DeputyUid',
+        'DeputyFirstname',
+        'DeputySurname',
+        'DeputyAddress1',
+        'DeputyAddress2',
+        'DeputyAddress3',
+        'DeputyAddress4',
+        'DeputyAddress5',
+        'DeputyPostcode',
+        'ReportType',
+        'MadeDate',
+        'OrderType',
+        'CoDeputy',
+        'Hybrid',
+    ];
+    
+    protected const UNEXPECTED_COLUMNS = [
+        'LastReportDay', 
+        'DeputyOrganisation'
+    ];
+
+    private array $processingOutput = [
+        'errors' => [],
+        'added' => 0,
+        'skipped' => 0,
+    ];
+    
+    private OutputInterface $cliOutput;
+
+    public function __construct(
+        private readonly S3Client $s3,
+        private readonly ParameterBagInterface $params,
+        private readonly LoggerInterface $logger,
+        private readonly CSVDeputyshipProcessing $csvProcessing,
+        private readonly PreRegistrationRepository $preReg,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void 
+    {
+        $this
+            ->setDescription('Process the Lay Deputies CSV from the S3 bucket')
+            ->addArgument('csv-filename', InputArgument::REQUIRED, 'Specify the file name of the CSV to retreive');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->cliOutput = $output;
+        $bucket = $this->params->get('s3_sirius_bucket');
+        $layReportFile = $input->getArgument('csv-filename');
+        $fileLocation = sprintf('/tmp/%s', $layReportFile);
+
+        try {
+            $this->s3->getObject([
+                'Bucket' => $bucket,
+                'Key' => $layReportFile,
+                'SaveAs' => $fileLocation
+            ]);
+        } catch (S3Exception $e) {
+            if (in_array($e->getAwsErrorCode(), S3Storage::MISSING_FILE_AWS_ERROR_CODES)) {
+                $logMessage = 'File %s not found in bucket %s';
+            } else {
+                $logMessage = 'Error retrieving file %s from bucket %s';
+            }
+            $logMessage = sprintf($logMessage, $layReportFile, $bucket);
+
+            $this->logger->error($logMessage);
+            $this->cliOutput->writeln(sprintf('%s - failure - %s', self::JOB_NAME, $logMessage));
+
+            return Command::FAILURE;
+        }
+
+        $data = $this->csvToArray($fileLocation);
+        if (count($data) >= 1 && $this->process($data) && empty($this->processingOutput["errors"])) {
+            if (!unlink($fileLocation)) {
+                $logMessage = sprintf('Unable to delete file %s.', $fileLocation);
+
+                $this->logger->error($logMessage);
+                $this->cliOutput->writeln(
+                    sprintf(
+                        '%s - partial - %s Output: %s',
```

Comment:
no such things as partial slack handler. it won't pick it up as a thing to alert on:

`pattern = r"^\w+(?:_\w+)* - (success|failure) - .+$"`